Refactor: Modular backend with services, routes, and cleaned structure

ASR1015 opened the pull request:
**Contributor Details -**
	•	Name: Anushka Srivastava
	•	GitHub ID: ASR1015
	•	Participation Program: GSSoC’25

 **Changes -** 
	•	Refactored the backend into a clean modular structure using services, routes, and handlers.
	•	Moved business logic from route files into a separate stock_service.py file.
	•	Added proper imports and ensured each module has a single responsibility.
	•	Cleaned and formatted the project structure for better scalability and maintainability.
	•	Tested the /api/stock?ticker=... endpoint and ensured valid, well-formatted JSON responses.

**Below are attached screenshots of successful output from /api/stock?ticker=TCS.NS:**
	•	JSON data response with chart structure
	•	Formatted historical stock data
	•	Paginated scroll results showing complete API behavior

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 52 AM" src="https://github.com/user-attachments/assets/1cf09298-d779-4031-83ea-8ac7b3e4fe0e" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 51 AM" src="https://github.com/user-attachments/assets/50fba112-ac9e-4ecf-99c4-d6645ee82b6d" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 49 AM" src="https://github.com/user-attachments/assets/7154caed-ec69-42b7-875e-c362115ed914" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 47 AM" src="https://github.com/user-attachments/assets/4de85462-c23a-45c8-bdd7-7d9463d902cf" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 46 AM" src="https://github.com/user-attachments/assets/a35d3640-4dd6-48ce-8a84-92f26506c7d5" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 44 AM" src="https://github.com/user-attachments/assets/02b8bdb8-4bb6-490b-b1e9-653aff2d2cce" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 41 AM" src="https://github.com/user-attachments/assets/7aede770-a9d6-4b48-91eb-29585b1ad0c9" />

<img width="1440" height="900" alt="Screenshot 2025-08-03 at 11 43 38 AM" src="https://github.com/user-attachments/assets/0125a6c3-c3e8-4a4d-a686-ea43d7a52b7c" />

**Files Changed -** 
	•	app.py
	•	app/routes/stock_routes.py
	•	app/services/stock_service.py
	•	app/services/stock_predict.py
	•	app/config.py

**How Has This Been Tested?**
	•	Ran Flask server locally on 127.0.0.1:10000
	•	Queried endpoint using /api/stock?ticker=TCS.NS
	•	Verified the structure, completeness, and accuracy of the output
	•	Confirmed the server runs with no errors or warnings

**Type of Change -** 
	•	Code refactor
	•	File structure cleanup
	•	Backend modularization
	•	Minor bug fix (import path errors)

**Checklist -** 
	•	My code follows the project’s guidelines
	•	I’ve tested the endpoint manually
	•	I’ve included relevant screenshots
	•	I’ve cleaned unused code/imports
	•	The backend runs successfully

**Everything is ready for review and merge. Thank you!**

SrigadaAkshayKumar commented:
@ASR1015 Please take a moment to understand the problem statement thoroughly. We need to add CSV files for testing purposes. Instead of directly fetching stock data using yfinance, we are using an Airflow scheduler to retrieve data periodically and store it in CSV files (for now).

You are expected to:

Add 2–3 sample CSV files.

Connect those CSV files to the backend using Python.

When a user selects a stock, the app should display data from the corresponding CSV — not by fetching it again from yfinance.

Please refer to Issue #62 and go through the description carefully to fully understand the requirements.

If you have any questions or need clarification, feel free to reach out. If you're not interested in continuing with this task, please let us know so that we can assign it to someone else.

Note:

Please remove all __pycache__ folders from the project (and add them to .gitignore).

Make sure to wrap all related files inside the backend/ folder; do not place any project folders outside of it.

ASR1015 commented:
Thank you for the clarification, @SrigadaAkshayKumar!
I’ve understood the updated requirements. I’ll proceed to:
	•	Create 2–3 sample CSV files with mock stock data.
	•	Connect those CSVs to the backend and serve data based on user selection.
	•	Ensure everything is organized within the backend/ folder and remove all __pycache__ directories.

I’ll update the PR accordingly. Please let me know if there’s anything else you’d like me to keep in mind!

ASR1015 commented:
Hi @SrigadaAkshayKumar 👋,

All requested changes have been implemented as per your detailed instructions:
	• Added 3 sample CSVs (TCS.NS, INFY.NS, RELIANCE.NS) and connected them to the backend using Python.
	• The backend now reads from local CSVs instead of fetching from yfinance.
	• Prediction logic is fully functional and uses Linear Regression over historical data from the CSVs.
	• Cleaned up all __pycache__ folders and added to .gitignore.
	• All relevant files are organized under the backend/ directory nothing placed outside.
	• CSVs were successfully generated and saved under backend/data/.
	• Functional testing of the /api/stock?ticker=... endpoint works correctly, with valid predictions and return projections.

SrigadaAkshayKumar commented:
@ASR1015 could you please share your LinkedIn profile URL and username also so I can mention you in the post?

also your name and email as soon as possible so that i can able to provide gssoc points and merge pr

ASR1015 commented:
Thank you so much! I’m incredibly grateful and excited to contribute to this amazing project! 

Here are my details for the post and GSSoC points:
Name: Anushka Srivastava
GitHub Username: https://github.com/ASR1015
LinkedIn Profile: https://www.linkedin.com/in/srivastavanushka/
Email: asrockstar555@gmail.com

Looking forward to more such meaningful contributions!
